### PR TITLE
Set HA version on WebSocket

### LINKED ISSUE #1999: Unhanded promise rejections depreciated
Please include the related YAML code:

```YAML
your:
   - yaml
```

Find my Home Assistant Configuration here:
[Link to Home Assistant Configuration](<link to repo here>)

Extension runs in:
- [X] VS Code for Desktop 
- [ ] Hassio Add-on 
- [ ] Visual Studio Online
- [ ] Other: 

I'm running VS Code on:
- [X] Windows 
- [ ] Mac 
- [ ] Linux 

I'm accessing my files:
- [ ] From local disk 
- [ ] Via remote SSH 
- [X] Smb
- [ ] Other:  

I see the following warning using v1.27:

Untitled2

### LINKED ISSUE #2011: V1.27 has connection problems with HomeAssistant in docker at least
Going back to V1.26 solves it

### LINKED ISSUE #2024: 1.27 crashes determining version
1.27 crashes when trying to run against **2022.3.5** both locally and over SSH.  Rolling back to 1.26 resolves the problem.

Please include the related YAML code:
No relevant yaml.  The extension crashes upon startup / retrieving data from HA

Find my Home Assistant Configuration here:
None

Extension runs in:
- [X] VS Code for Desktop 
- [ ] Hassio Add-on 
- [ ] Visual Studio Online
- [ ] Other: 

I'm running VS Code on:
- [ ] Windows 
- [X] Mac 
- [ ] Linux 

I'm accessing my files:
- [X] From local disk 
- [X] Via remote SSH 
- [ ] Smb
- [ ] Other:  

Relevant error logs.
```
/home/felix/.vscode-server/extensions/keesschollaart.vscode-home-assistant-1.27.0/node_modules/home-assistant-js-websocket/dist/haws.cjs:508
    const [haMajor, haMinor, haPatch] = version.split(".", 3);
                                                ^

TypeError: Cannot read properties of undefined (reading 'split')
    at atLeastHaVersion (/home/felix/.vscode-server/extensions/keesschollaart.vscode-home-assistant-1.27.0/node_modules/home-assistant-js-websocket/dist/haws.cjs:508:49)
    at entitiesColl (/home/felix/.vscode-server/extensions/keesschollaart.vscode-home-assistant-1.27.0/node_modules/home-assistant-js-websocket/dist/haws.cjs:1010:32)
    at Object.subscribeEntities (/home/felix/.vscode-server/extensions/keesschollaart.vscode-home-assistant-1.27.0/node_modules/home-assistant-js-websocket/dist/haws.cjs:1013:47)
    at /home/felix/.vscode-server/extensions/keesschollaart.vscode-home-assistant-1.27.0/out/language-service/src/home-assistant/haConnection.js:61:20
    at new Promise (<anonymous>)
    at HaConnection.getHassEntities (/home/felix/.vscode-server/extensions/keesschollaart.vscode-home-assistant-1.27.0/out/language-service/src/home-assistant/haConnection.js:55:33)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async HaConnection.getEntityCompletions (/home/felix/.vscode-server/extensions/keesschollaart.vscode-home-assistant-1.27.0/out/language-service/src/home-assistant/haConnection.js:165:26)
    at async HomeAssistantLanguageService.getServiceAndEntityCompletions (/home/felix/.vscode-server/extensions/keesschollaart.vscode-home-assistant-1.27.0/out/language-service/src/haLanguageService.js:174:48)
    at async HomeAssistantLanguageService.onCompletion (/home/felix/.vscode-server/extensions/keesschollaart.vscode-home-assistant-1.27.0/out/language-service/src/haLanguageService.js:121:43)
    ```

### PR DESCRIPTION
**Pull request by frenck**
Most likely resolves:

- fixes #2024
- fixes #2011
- fixes #1999
- fixes https://github.com/hassio-addons/addon-vscode/issues/421
- fixes https://github.com/hassio-addons/addon-vscode/issues/419

### CODE REVIEW

File: src/language-service/src/home-assistant/socket.ts [hunk 1]
```diff
@@ -16,6 +16,10 @@ const MSG_TYPE_AUTH_OK = "auth_ok";
 const ERR_CANNOT_CONNECT = 1;
 const ERR_INVALID_AUTH = 2;
 
+export interface HaWebSocket extends WebSocket {
+  haVersion: string;
+}
+
```
Comment (ludeeus) [inline src/language-service/src/home-assistant/socket.ts:22]: `import type { HaWebSocket } from "home-assistant-js-websocket/dist/socket"` ?

Comment (frenck) [reply src/language-service/src/home-assistant/socket.ts:22]: Doh 🤦 Thanks!

**Review by ludeeus — APPROVED** [review]
LGTM

**Comment by michelsup** [comment]
Well, for me it hasn't solved the #419 issue. There is no more error but the identity completion still doesn't work.

**Pull request merged**